include/dlaf/mc/cholesky.h

```diff
@@ -0,0 +1,309 @@
+                      dlaf::comm::sync::broadcast::receive_from(index, comm_wrapper().colCommunicator(),
+                                                                tile);
+                      return std::move(tile);
+                    }),
+                k_rank_row, mat.tileSize(GlobalTileIndex(k, k)), serial_comm());
+          }
+        }
+
+        for (SizeType i_local = distr.nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+             i_local < localnrtile_rows; ++i_local) {
+          // Update panel mat(i,k) with trsm (blas operation), using data mat.read(k,k)
+          hpx::dataflow(matrix_HP_executor, hpx::util::unwrapping(tile::trsm<T, Device::CPU>),
+                        blas::Side::Right, uplo, blas::Op::ConjTrans, blas::Diag::NonUnit, 1.0, kk_tile,
+                        std::move(mat(LocalTileIndex{i_local, k_local_col})));
+
+          // Avoid useless communications if one-row communicator grid
+          if (row_comm_size > 1) {
+            // Broadcast the panel row-wise
+            hpx::dataflow(hpx::util::unwrapping([](auto&& tile, auto&& comm_wrapper) {
+                            dlaf::comm::sync::broadcast::send(comm_wrapper().rowCommunicator(), tile);
+                          }),
+                          mat.read(LocalTileIndex{i_local, k_local_col}), serial_comm());
+          }
+
+          panel[i_local] = mat.read(LocalTileIndex{i_local, k_local_col});
+        }
+      }
+      else {
+        for (SizeType i_local = distr.nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+             i_local < localnrtile_rows; ++i_local) {
+          auto i = distr.globalTileFromLocalTile<Coord::Row>(i_local);
+
+          // Avoid useless communications if one-row communicator grid
+          if (row_comm_size > 1) {
+            // Update the panel row-wise
+            panel[i_local] = hpx::dataflow(  //
+                hpx::util::unwrapping(
+                    [](auto index, auto&& tile_size, auto&& comm_wrapper) -> Tile<const T, Device::CPU> {
+                      memory::MemoryView<T, Device::CPU> mem_view(
+                          util::size_t::mul(tile_size.rows(), tile_size.cols()));
+                      Tile<T, Device::CPU> tile(tile_size, std::move(mem_view), tile_size.rows());
+                      dlaf::comm::sync::broadcast::receive_from(index, comm_wrapper().rowCommunicator(),
+                                                                tile);
+                      return std::move(tile);
+                    }),
+                k_rank_col, mat.tileSize(GlobalTileIndex(i, k)), serial_comm());
+          }
+        }
+      }
+
+      for (SizeType j_local = distr.nextLocalTileFromGlobalTile<Coord::Col>(k + 1);
+           j_local < localnrtile_cols; ++j_local) {
+        // Choose "high priority" for first tile of the trailing matrix
+        auto trailing_matrix_executor = (j_local == k + 1) ? matrix_HP_executor : matrix_normal_executor;
```

Comment:
This line is wrong. you are comparing a local index (`j_local`) with a global index `k+1`:
The option are two in this case:
Either only the global k+1 column is high priority, or the next column of each rank is high priority. We should check which performs best.